src/app/core/components/viewchartarea/viewchartarea.component.spec.ts

```diff
@@ -0,0 +1,93 @@
+import 'jest-canvas-mock';
+import {
+  createComponentFactory, Spectator,
+} from '@ngneat/spectator/jest';
+import { ChartData } from 'chart.js';
+import { ViewChartAreaComponent } from './viewchartarea.component';
+
+describe('ViewChartAreaComponent', () => {
+  /*
+  * Component Setup
+  * */
+
+  let spectator: Spectator<ViewChartAreaComponent>;
+
+  const createComponent = createComponentFactory({
+    component: ViewChartAreaComponent,
+    detectChanges: false,
+  });
+
+  /*
+  * Generate Mock Data
+  * */
+
+  const generateDatasets = (dataSources: number, dataPoints: number): any[] => {
+    const datasets = [];
+
+    for (let i = 0; i < dataSources; i++) {
+      const data = [];
+
+      for (let x = 0; x < dataPoints; x++) {
+        const dataPoint = Math.floor(Math.random() * 100);
+        data.push(dataPoint);
+      }
+      const dataset = {
+        label: 'Item ' + i.toString(),
+        data,
+      };
+
+      datasets.push(dataset);
+    }
+
+    return datasets;
+  };
+
+  const generateChartData = (dataSources: number, dataPoints: number): ChartData => {
+    const datasets = generateDatasets(dataSources, dataPoints);
+    const data: ChartData = {
+      labels: [],
+      datasets,
+    };
+
+    return data;
+  };
+
+  /*
+   * Test Setup
+   * */
+
+  beforeEach(() => {
+    spectator = createComponent();
+  });
+
+  /*
+   * Test Methods
+   * */
+
+  it('should not handle more than 8 data points', () => {
+    const data: ChartData = generateChartData(9, 2);
+
+    expect(spectator.component.maxSources).toBe(8);
+    expect(() => spectator.component.render(data)).toThrow();
+  });
+
+  it('should render chart when data arrives', () => {
+    const data: ChartData = { labels: [], datasets: [] };
+    spectator.setInput('data', data);
+
+    // Manually trigger change detection
+    spectator.component.ngOnChanges({
+      data: {
+        currentValue: data,
+        previousValue: null,
+        firstChange: true,
+        isFirstChange: () => true,
+      },
+    });
+
+    // Make sure expected values are present after input is set
+    expect(spectator.component.data).toBeTruthy();
+    expect(spectator.component.canvas).toBeTruthy();
+    expect(spectator.component.chart).toBeTruthy();
```

Comment:
Would be good to test that not only _some_ canvas is there, but also that the canvas/chart has the same data that was passed in component on line 76.